files/aiida-core-constraints.py

```diff
@@ -0,0 +1,37 @@
+#!/usr/bin/env python
+"""Script to convert aiida setup.json to pip constraints file.
```

Comment:
It's not constraints but rather requirements. You'd usually use https://github.com/jazzband/pip-tools to generate the pinned env deps aka constraints.

Reply:
Also, pip-tools can generate pins from `setup.py` natively so you could just replace this script with an existing tool: https://github.com/jazzband/pip-tools#requirements-from-setuppy

Reply:
no that creates a lock file, with strict version pinning. we just want a constraints file, for when you run subsequent pip installs, i.e. these installs can change the version of packages if necessary, but only within the range specified by setup.json

Reply:
@chrisjsewell true but the purpose of [constraints file in pip](https://pip.pypa.io/en/latest/user_guide/#constraints-files) is exactly that it should be a lock file. So that the usage would be `pip install -r requirements.txt -c constraints.txt`. The former should usually contain open-ended direct dependencies while the latter would pinpoint what exactly is allowed (it's best when there's also `--require-hashes` too) making the virtualenv setup reproducible.